serializers/shared/src/main/java/io/pravega/schemaregistry/serializer/shared/credentials/PravegaCredentialProvider.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializer.shared.credentials;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.impl.Credentials;
+import io.pravega.schemaregistry.common.CredentialProvider;
+import lombok.AllArgsConstructor;
+
+/**
+ * Credential provider that is a wrapper over and uses pravega credentials to authenticate to schema registry.  
+ */
+@AllArgsConstructor
+public class PravegaCredentialProvider implements CredentialProvider {
+    private final Credentials credentials;
+
+    public PravegaCredentialProvider() {
+        credentials = ClientConfig.builder().build().getCredentials();
```

Comment:
So, this will pick credentials from the environment too right? Is that intended?

Java system properties:
```
pravega.client.auth.method
pravega.client.auth.token
```
Env. variables:

```
pravega_client_auth_method
pravega_client_auth_token
```
One issue (or rather feature) I see with this is that if a Pravega client and Schema Registry is running on the same, they'll both obtain the same credentials from the environment/system.

Reply:
Users need to pass the credential provider to schema registry client. 
case 1: separate creds for schema regsitry
they load it explicitly

case 2: they want to use pravega credentials which were not loaded automatically. 
they create `PravegaCredentialProvider` instance with the constructor with pravega client config

case 3: they simply want to load the credentials from the environment without passing the client config, they could simply create the constructor with no args, which creates a default client config instance which loads the credentials from the environment.